### ISSUE TITLE
the test intext_par.ml is flaky

### ISSUE THREAD

**Issue by gasche** [ocaml/ocaml#12421]
Our CI observes a rare intermittent failure of the test intext_par.ml (deserialization running in parallel).

An intext_par.ml failure was observed in the following issues: #12320, #12029, #12058.

On both #12320 and #12029 the failure was observed on intext_par.opt (the native-compiled binary) on the macos CI machine, with the following error message:
```
Fatal error: Fatal error during mutex_free: Resource busy
```

**Comment by gasche** [comment]
(I wonder if this is something that @fabbing or @OlivierNicole would be curious to look at; they had so much fun chasing obscure TSan test failures on macos that they may have experience by now, or know someone who does.)

**Comment by OlivierNicole** [comment]
Personally I can add it to my todo-list but it won’t be among my top priorities at the moment. A brief run with TSan with deadlock inversion detection enabled shows nothing obvious on Linux, but I haven’t tried on macOS.

**Comment by xavierleroy** [comment]
Not only this test is flaky, but it uses 8 domains, causing it to run slowly on most of our CI machines (2-core VMs) and also to run out of memory on one of our CI machines (a 1Gb rRPi3 running FreeBSD).

I direct-pushed 7f480854c0 to make it conform to the `OCAML_TEST_SIZE` protocol: https://github.com/ocaml/ocaml/blob/7f480854c01d6726d02e31c2b7635967b904703d/testsuite/HACKING.adoc?plain=1#L120-L140

For the purpose of reproducing the flakiness, setting `OCAML_TEST_SIZE=3` gives back the original 8-domain behavior.

**Comment by xavierleroy** [comment]
```
Fatal error: Fatal error during mutex_free: Resource busy
```
Note that this could be a case of #12445: we have a bunch of domains that close out_channels concurrently, so a GC could occur and a channel be finalized while it is locked and no longer reachable from OCaml roots.

**Comment by OlivierNicole** [comment]
Now that #12445 is merged, is the test still flaky? I do not see it failing when checking on the last dozen  of CI runs.

**Comment by gasche** [comment]
Whenever I personally observe a flaky result for a test, I update the corresponding issue, so the answer is "I don't know about a failure after 12445". Note that the failures were still observed somewhat rarely :  three failures between 12058 and 12445, so about 125 PRs by failure, and we only opened 100 PRs since 12445. (And even less since it was merged.)

We are very likely to forget to ever close this issue if the failure does not show up again, so it is probably sensible -- we have good suspicions that 12445 fixed it -- to close now, and reopen only if it shows up again. Let's do that.
